### PR TITLE
auto update indicator in the atom://about page

### PR DESCRIPTION
**Pull request by ethanluoyc**
I am just starting on this so it is not yet finished. (A long list of features to add and I am new to open source )

about

This is a first step to merge the atom/release-notes and atom/about according to atom/atom#9164. This is currently only a mockup for how the about page can possibly evolve.

I will try to finish the rest as soon as possible, but I do hope to have some feedback on my current progress.

atom/release-notes#38 solved the problem of the squirrel, which I think should also be patched here. So I will have to wait for the PR to be merged too before I can update on the squirrel icon problem as mentioned in https://github.com/atom/atom/issues/6922. I will also probably merge the tests of the two packages.

### CODE REVIEW

**Comment by ethanluoyc** [comment]
One test is failing because the behavior of the about page is changed in this PR

**Comment by ethanluoyc** [comment]
@mnquintana you can finish the work in atom/about#1 and then I can finish the UI part. What I need is and event emitted that I can capture to display the "updates available".

**Comment by simurai** [comment]
Is the "Check for Updates" button needed? It might give the impression that there is no auto-update and you have to do it manually each time. "Check for updates on open" #1 could be an alternative.

**Comment by jerone** [comment]
[**@simurai**](https://github.com/simurai) commented on [19 nov. 2015 11:07 CET](https://github.com/atom/about/pull/5#issuecomment-158011191 "2015-11-19T10:07:29Z - Replied by Github Comment Enhancer"):
> Is the "Check for Updates" button needed? It might give the impression that there is no auto-update and you have to do it manually each time. "Check for updates on open" [#1](https://github.com/atom/about/issues/1 "Check for updates on open") could be an alternative.

That's how all browser are doing it (Firefox, Chrome). With that change the "Check for Update" main menu item should also be removed.

**Comment by 50Wliu** [comment]
I like @jerone's suggestion of keeping it in the About page and removing the option from the menu.

**Comment by mnquintana** [comment]
I personally think it's not great UX to remove Check for Updates from the menu entirely - rather, I'd prefer to keep it, but instead remap it to open the About dialog and check for updates there.

**Comment by ethanluoyc** [comment]
i agree with that.



—

File: lib/about-view.coffee [hunk 1]
```diff
@@ -73,6 +81,27 @@ class AboutView extends ScrollView
     @on 'click', '.metrics-open', ->
       atom.workspace.open('atom://config/packages/metrics')
 
+  checkForUpdates: () =>
```
Comment (benogle) [inline lib/about-view.coffee]: This should use the [AutoUpdateManager](https://github.com/atom/atom/blob/master/src/browser/auto-update-manager.coffee). There may need to be code changes to expose more information to the renderer process.

Comment (ethanluoyc) [inline lib/about-view.coffee]: @benogle I am not sure what you mean by this, how do I require the AutoUpdateManager if it is from the core?

**Comment by abe33** [comment]
@ethanluoyc I think that's why @benogle said there may need to change the code in core to expose more information, for instance so that you can register to the [auto-updater events](https://github.com/atom/atom/blob/8fb4e887df800438b2deeb23a2c23004b9ff358a/src/browser/application-menu.coffee#L15) instead of reimplementing the update process.

**Comment by ethanluoyc** [comment]
@abe33 I got it now, I will wait for that to finish.
Actually I wonder whether they should open up auto-updater as a separate package which will be shipped with Atom? Similar to the about package

**Comment by abe33** [comment]
> Actually I wonder whether they should open up auto-updater as a separate package which will be shipped with Atom?

It's seems highly improbable because, AFAIK, it's a feature provided by Electron and not Atom itself. And I think the `AutoUpdateManager` is used in the main process (and not in renderer processes).

**Comment by ethanluoyc** [comment]
Okay I am not familiar with Electron but I guess this makes a good point

**Pull request closed without merging**